### ISSUE TITLE
Theme colors of Universal Login behave unexpectedly after import

### ISSUE THREAD

**Issue by ramirocarra** [auth0/auth0-deploy-cli#828]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/auth0-deploy-cli#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://github.com/auth0/auth0-deploy-cli#documentation) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-deploy-cli/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/auth0-deploy-cli/releases/latest) of this tool and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

After importing the configuration, the theme colors seem to kind of "reset" to black and white. The behavior is very strange, as when you go to the management console you can see the correct configuration, but when clicking on "try" the colors do not match with the configuration. (This suggests there is also a bug in the server)

Changing any parameter and clicking "save and publish" solves the issue, until a new import when breaks again in the same way.

I could reproduce it on versions 17.18.0 and 17.17.7, but the issue is not present on v17.17.6 which executes a correct import with no issues.

Note that this happens even if you are importing the same current configuration with no changes

### Expectation

Theme configuration should be applied correctly on each import and there should be no changes in the Universal Login if the configuration does not change

### Reproduction

1. Given that the Theme has been configured with the console, the Universal Login page works correctly
2. Getting the configuration with `a0deploy export` and applying it again using `a0deploy import`
3. After import the Universal login Page has a different color configuration (black and white) than the one imported. The Auth0 management console shows correctly the configuration imported, but it does not match the actual view of the page.
4. After doing any changes and clicking "save and publish" the page works correctly again
5. Running `a0deploy import` again breaks the page and behavior is repeated from step 3

My specific theme config:
```
{
  "colors": {
    "primary_button": "#1d4cae",
    "primary_button_label": "#ffffff",
    "secondary_button_border": "#c9cace",
    "secondary_button_label": "#ffffff",
    "base_focus_color": "#1d4cae",
    "base_hover_color": "#15181f",
    "links_focused_components": "#1d4cae",
    "header": "#ffffff",
    "body_text": "#ffffff",
    "widget_background": "#15181f",
    "widget_border": "#c9cace",
    "input_labels_placeholders": "#65676e",
    "input_filled_text": "#ffffff",
    "input_border": "#c9cace",
    "input_background": "#15181f",
    "icons": "#65676e",
    "error": "#d03c38",
    "success": "#13a688"
  },
  "fonts": {
    "font_url": "https://fonts.cdnfonts.com/css/roboto",
    "reference_text_size": 16,
    "title": {
      "bold": false,
      "size": 150
    },
    "subtitle": {
      "bold": false,
      "size": 87.5
    },
    "body_text": {
      "bold": false,
      "size": 87.5
    },
    "buttons_text": {
      "bold": false,
      "size": 100
    },
    "input_labels": {
      "bold": false,
      "size": 100
    },
    "links": {
      "bold": true,
      "size": 87.5
    },
    "links_style": "normal"
  },
  "borders": {
    "button_border_weight": 1,
    "buttons_style": "rounded",
    "button_border_radius": 3,
    "input_border_weight": 1,
    "inputs_style": "rounded",
    "input_border_radius": 3,
    "widget_corner_radius": 5,
    "widget_border_weight": 0,
    "show_widget_shadow": false
  },
  "widget": {
    "logo_position": "center",
    "logo_url": "SENSORED",
    "logo_height": 57,
    "header_text_alignment": "center",
    "social_buttons_layout": "bottom"
  },
  "page_background": {
    "page_layout": "center",
    "background_color": "#3c4a71",
    "background_image_url": ""
  },
  "displayName": "Unnamed Theme"
}
```

### Deploy CLI version

17.18.0

### Node version

18.16.0

**Comment by willvedd** [comment]
Hi @ramirocarra thanks for opening this issue. I'm unable to reproduce, my theme settings get applied as normal. Are you able to try again? This may have been an intermittent server-side issue and I don't really see anything in the releases after 7.17.6 that would explain the issue.

**Comment by poolieweb** [comment]
We have the same issue, I suspect that we have used a flag that has been removed in new versions of the cli and we cant overwrite. We lose the custom styling when importing and the identifier first selection. If you go to the console and for custom styling change one property and save we get the full contents of the import.

**Comment by ramirocarra** [comment]
Hi @willvedd! thanks for your response. I created a tenant just to test this and I was able to reproduce it with version 7.18.0 and a minimalistic configuration (Removed most of my specific config, Clients, etc, and left only branding, themes, and a few more settings)

**Comment by poolieweb** [comment]
@willvedd can I ask when exporting the config from your tenant do you get any warnings about tenant flags ?
We are on the new UK tenant and get the following :
`warn: The following tenant flags have not been updated because deemed incompatible with the target tenant: allow_changing_enable_sso, disable_impersonation, new_universal_login_experience_enabled, universal_login, dashboard_new_onboarding
      These flags can likely be removed from the tenant definition file. If you believe this removal is an error, please report via a Github issue.`
  Which is strange as this is the export. if we import the export with no changes we get no errors, but the styling get back defaulted to standard style and user name and password flow, rather than the choices we made before.

**Comment by poolieweb** [comment]
To add the following properties are being missed when importing:
```
  identifier_first: true
  universal_login_experience: new
```
They have to be reapplied in the console

**Comment by willvedd** [comment]
@ramirocarra Thanks for the clarification. I can again attempt to recreate with the config that you've provided above. And to clarify, you're not seeing _any_ of the colors in the theme being applied on import? Or is it just when you click the "try" button?

@poolieweb That is a separate issue for this one. Those tenant flags that you listed have been deprecated and the appropriate settings can be found elsewhere. I'd be happy to address your issues and concerns specifically in a separate issue.

**Comment by ramirocarra** [comment]
@willvedd In the console everything looks fine, I can see the applied config and even the preview seems fine. Is only when I go to the real page, by clicking "Try" or logging in through my app, that I see only back and whites

**Comment by poolieweb** [comment]
@ramirocarra I agree the tenant tags could be a different issue, but we get the same issue as @willvedd as well. It like there is a shadow property on the server we can't updates as the CLI has moved on, but a refresh in the console overrides. 

What's the best way to split this out ? new ticket ?

**Comment by willvedd** [comment]
@ramirocarra Thanks for your patience here. This is indeed a bug on the server. In short, there is some internal logic that determines if a theme exists. It is possible to exhibit the bug you described if theme and branding settings are modified in a specific order. And in fact, a re-ordering of the branding handler was introduced after 7.17.6. 

We will be working to fix this in the meantime. Hopefully have something for you soon!

**Comment by sergiught** [comment]
Hey folks, I was able to reproduce this. The issue you're experiencing is caused by the fact that the cache does not invalidate once a theme gets created after branding settings get applied. 

When using the UI and clicking "Save and Publish" the cache is forcefully invalidated, but that does not happen through the API call that the deploy-cli leverages. 

This is definitely something that needs to be fixed on the API side, however we can also readjust the order of the API calls within the deploy cli to create/update the theme first and then apply the branding settings as that will also invalidate the cache. 

The cache lasts just 5 minutes, afterwards the theme gets displayed properly. 

We'll leave this issue open until we can both:

- Invalidate the cache on create/update of the theme on the API side
- Create/Update the theme first, and then apply any other branding settings on the Deploy CLI

**Comment by willvedd** [comment]
What @sergiught above, this issue is a server-side cache invalidation issue. It will resolve itself in five minutes. However, we released a fix in [7.19.0](https://github.com/auth0/auth0-deploy-cli/releases/tag/v7.19.0) that should prevent this issue from occurring at all. Thanks for everyone's help in troubleshooting this one!
